compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/KnownTargets.cpp

```diff
@@ -215,48 +235,147 @@ std::optional<TargetDetails> getAMDGPUTargetDetails(StringRef target) {
+    return target;
+
   return llvm::StringSwitch<StringRef>(target.lower())
       .Case("mi300x", "gfx942")
       .Case("mi300a", "gfx940")
       .Cases("mi250x", "mi250", "mi210", "cdna2", "gfx90a")
+      .Case("cdna1", "gfx908")
       .Cases("rx7900xtx", "rx7900xt", "gfx1100")
       .Cases("rx7800xt", "rx7700xt", "gfx1101")
       .Default(StringRef());
 }
 
+//===----------------------------------------------------------------------===//
+// Known Apple target details
+//===----------------------------------------------------------------------===//
+
+std::optional<TargetDetails> getAppleTargetDetails() {
+  ComputeBitwidths computeBitwdiths =
+      allIntComputeBits | ComputeBitwidths::FP32 | ComputeBitwidths::FP16;
+  // clang-format off
+  static const WgpDetails wgp = {
+      computeBitwdiths,   allStorageBits,     allSubgroupOps,  allDotProductOps,
+      /*mmaCount=*/0,     /*mmaOps=*/nullptr, {32, 32},
+      {1024, 1024, 1024}, 1024,               32 * 1024};
+  // clang-format on
+
+  return TargetDetails{&wgp, nullptr};
+}
+
+//===----------------------------------------------------------------------===//
+// Known ARM target details
+//===----------------------------------------------------------------------===//
+
+const WgpDetails *getValhallWgpDetails() {
+  ComputeBitwidths computeBitwdiths =
+      allIntComputeBits | ComputeBitwidths::FP32 | ComputeBitwidths::FP16;
+  // clang-format off
+  static const WgpDetails valhallWgp = {
+      computeBitwdiths,   allStorageBits,     allSubgroupOps,  allDotProductOps,
+      /*mmaCount=*/0,     /*mmaOps=*/nullptr, {16},            {512, 512, 512},
+      512,                32 * 1024};
+  // clang-format on
+  return &valhallWgp;
+}
+
+std::optional<TargetDetails> getARMGPUTargetDetails(StringRef target) {
+  const WgpDetails *valhallWgp = getValhallWgpDetails();
+
+  // Note that the underlying GPU may have certain capabilities but the Android
+  // version and driver stack may not expose them. So the following is just and
+  // will always be approximate.
+
+  return llvm::StringSwitch<std::optional<TargetDetails>>(target.lower())
+      // Mali-G715: https://vulkan.gpuinfo.org/displayreport.php?id=29754
+      .Cases("mali-g715", "mali-g615", "valhall4",
+             TargetDetails{valhallWgp, nullptr})
+      // Mali-G710: https://vulkan.gpuinfo.org/displayreport.php?id=30471
+      .Cases("mali-g710", "mali-g510", "mali-g310", "valhall3",
+             TargetDetails{valhallWgp, nullptr})
+      // Mali-G78: https://vulkan.gpuinfo.org/displayreport.php?id=29994
+      .Cases("mali-g78", "valhall2", TargetDetails{valhallWgp, nullptr})
+      // Mali-G57: https://vulkan.gpuinfo.org/displayreport.php?id=24636
+      .Cases("mali-g77", "mali-g57", "valhall1", "valhall",
+             TargetDetails{valhallWgp, nullptr})
+      .Default(std::nullopt);
+}
+
+StringRef normalizeARMGPUTarget(StringRef target) {
+  if (target == "valhall")
+    return "valhall1";
```

Comment:
Is this intentional?

Reply:
Yes. We don't have a formal established target naming scheme for arm (from llvm or something) so i'm just making it up for our own uses. Trying to be consistent here to use `valhallN` like `rdnaN` or `cdnaN`. If people give just `valhall` we default to the lowest level of support there.